__tests__/integration/hathorwallet_others.test.js

```diff
@@ -0,0 +1,1247 @@
+      .toThrow('utxos to fill');
+  });
+});
+
+describe('markUtxoSelected', () => {
+  /** @type HathorWallet */
+  let hWallet;
+  /** @type string */
+  let txHash;
+  /** @type number */
+  let oIndex;
+
+  beforeAll(async () => {
+    hWallet = await generateWalletHelper();
+    const { hash } = await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 10);
+    const { index } = hWallet.getUtxos().utxos[0];
+    txHash = hash;
+    oIndex = index;
+  });
+
+  afterAll(async () => {
+    hWallet.stop();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  /*
+   * We will validate this method's results by the following checks:
+   * 1 - getTx() - direct access to the wallet full history
+   * 2 - getUtxos()
+   * 3 - getAllUtxos()
+   * 4 - getUtxosForAmount()
+   */
+
+  it('should mark utxos as selected', async () => {
+    // Validating utxo current state as not selected
+    let rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    loggers.test.log('Output before', rawOutput);
+    expect(rawOutput.selected_as_input).toStrictEqual(false);
+
+    // Marking it as selected
+    hWallet.markUtxoSelected(txHash, oIndex, true);
+
+    // Validation 1
+    rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    loggers.test.log('Output after', rawOutput);
+    expect(rawOutput.selected_as_input).toStrictEqual(true);
+
+    // Validation 2
+    expect(hWallet.getUtxos()).toMatchObject({
+      total_utxos_available: 0,
+      utxos: [expect.objectContaining({
+        tx_id: txHash,
+        locked: true,
+      })],
+    });
+
+    // Validation 3
+    const utxosGenerator = hWallet.getAllUtxos();
+    expect(utxosGenerator.next()).toStrictEqual({
+      value: undefined,
+      done: true,
+    });
+
+    // Validation 4
+    expect(() => hWallet.getUtxosForAmount(10)).toThrow('utxos to fill');
+  });
+
+  it('should mark utxos as not selected', async () => {
+    // Validating utxo current state as selected
+    let rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    expect(rawOutput.selected_as_input).toStrictEqual(true);
+
+    // Marking it as not selected
+    hWallet.markUtxoSelected(txHash, oIndex, false);
+
+    // Validation 1
+    rawOutput = hWallet.getTx(txHash).outputs[oIndex];
+    expect(rawOutput.selected_as_input).toStrictEqual(false);
+
+    // Validation 2
+    expect(hWallet.getUtxos()).toMatchObject({
+      total_utxos_available: 1,
+      utxos: [expect.objectContaining({
+        tx_id: txHash,
+        locked: false
+      })],
+    });
+
+    // Validation 3
+    const utxosGenerator = hWallet.getAllUtxos();
+    expect(utxosGenerator.next()).toStrictEqual({
+      value: expect.objectContaining({ txId: txHash }),
+      done: false,
+    });
+
+    // Validation 4
+    expect(hWallet.getUtxosForAmount(10)).toStrictEqual({
+      changeAmount: 0,
+      utxos: [expect.objectContaining({ txId: txHash })]
+    });
+  });
+});
+
+describe('consolidateUtxos', () => {
```

Comment:
I think we need 1 more test, trying to consolidate more than the max inputs allowed.
It should consolidate only the max allowed.

Reply:
This test will be implemented on the next PR, together with other validations to the max inputs allowed.